packages/vendure-plugin-sort-by-popularity/src/sort.service.ts

```diff
@@ -1,12 +1,180 @@
+    private channelService: ChannelService,
+    private collectionService: CollectionService
+  ) {}
+  async onModuleInit() {
+    this.jobQueue = await this.jobQueueService.createQueue({
+      name: 'calculate-scores',
+      process: async (job) => {
+        const channel = await this.channelService.getChannelFromToken(
+          job.data.channelToken
+        );
+        this.setProductPopularity(
+          RequestContext.deserialize(job.data.ctx),
+          channel.id
+        );
+      },
+    });
+  }
+
+  async setProductPopularity(
+    ctx: RequestContext,
+    channleId: ID
+  ): Promise<Success> {
+    const groupedOrderItems = await this.connection
+      .getRepository(ctx, OrderItem)
+      .createQueryBuilder('orderItem')
+      .innerJoin('orderItem.line', 'orderLine')
+      .select([
+        'count(product.id) as count',
+        'orderItem.line',
+        'orderLine.productVariant',
+        'orderLine.order',
+      ])
+      .innerJoin('orderLine.productVariant', 'productVariant')
+      .addSelect([
+        'productVariant.deletedAt',
+        'productVariant.enabled',
+        'productVariant.id',
+      ])
+      .innerJoin('orderLine.order', 'order')
+      .innerJoin('productVariant.product', 'product')
+      .addSelect(['product.deletedAt', 'product.enabled', 'product.id'])
+      .innerJoin('productVariant.collections', 'collection')
+      .addSelect(['collection.id'])
+      .innerJoin('order.channels', 'order_channel')
+      .andWhere('order.orderPlacedAt is NOT NULL')
+      .andWhere('product.deletedAt IS NULL')
+      .andWhere('productVariant.deletedAt IS NULL')
+      .andWhere('product.enabled')
+      .andWhere('productVariant.enabled')
+      .andWhere('order_channel.id = :id', { id: channleId })
+      .addGroupBy('product.id')
+      .addOrderBy('count', 'DESC')
+      .getRawMany();
+    const maxCount = groupedOrderItems[0].count;
+    const maxValue = 1000;
+    const productRepository = this.connection.getRepository(ctx, Product);
+    await productRepository.save(
+      groupedOrderItems.map((gols) => {
+        return {
+          id: gols.product_id,
+          customFields: {
+            popularityScore: Math.round((gols.count / maxCount) * maxValue),
+          },
+        };
+      })
+    );
+    await this.assignScoreValuesToCollections(ctx);
+    return { success: true };
+  }
+  async assignScoreValuesToCollections(ctx: RequestContext) {
+    const allCollectionsScores = await this.getEachCollectionsScore(ctx);
+    await this.addUpTheTreeAndSave(allCollectionsScores, ctx);
+  }
+
+  async getEachCollectionsScore(
+    ctx: RequestContext
+  ): Promise<{ id: string; score: number }[]> {
+    const collectionsRepo = this.connection.getRepository(ctx, Collection);
+    const productsRepo = this.connection.getRepository(ctx, Product);
+    const allCollectionIds = await collectionsRepo
+      .createQueryBuilder('collection')
+      .select('id')
+      .getRawMany();
```

Comment:
I think this function is not channel aware right now? Shouldn't we only get collections for the current selection? Like you do when querying orders.